pupil_src/launchables/world.py

```diff
@@ -325,15 +325,63 @@ def get_timestamp():
             ("System_Graphs", {}),
         ]
 
+        def consume_events_and_render_buffer():
+            gl_utils.glViewport(0, 0, *camera_render_size)
+            for p in g_pool.plugins:
+                p.gl_display()
+
+            gl_utils.glViewport(0, 0, *window_size)
+            try:
+                clipboard = glfw.glfwGetClipboardString(main_window).decode()
+            except AttributeError:  # clipboard is None, might happen on startup
+                clipboard = ""
+            g_pool.gui.update_clipboard(clipboard)
+            user_input = g_pool.gui.update()
+            if user_input.clipboard != clipboard:
+                # only write to clipboard if content changed
+                glfw.glfwSetClipboardString(main_window, user_input.clipboard.encode())
+
+            for button, action, mods in user_input.buttons:
+                x, y = glfw.glfwGetCursorPos(main_window)
+                pos = glfw.window_coordinate_to_framebuffer_coordinate(
+                    main_window, x, y, cached_scale=None
+                )
+                pos = normalize(pos, camera_render_size)
+                # Position in img pixels
+                pos = denormalize(pos, g_pool.capture.frame_size)
+
+                for plugin in g_pool.plugins:
+                    if plugin.on_click(pos, button, action):
+                        break
+
+            for key, scancode, action, mods in user_input.keys:
+                for plugin in g_pool.plugins:
+                    if plugin.on_key(key, scancode, action, mods):
+                        break
+
+            for char_ in user_input.chars:
+                for plugin in g_pool.plugins:
+                    if plugin.on_char(char_):
+                        break
+
+            glfw.glfwSwapBuffers(main_window)
+
         # Callback functions
         def on_resize(window, w, h):
             nonlocal window_size
             nonlocal camera_render_size
-            nonlocal hdpi_factor
+            nonlocal content_scale
             if w == 0 or h == 0:
                 return
-            hdpi_factor = glfw.getHDPIFactor(window)
-            g_pool.gui.scale = g_pool.gui_user_scale * hdpi_factor
+
+            # Always clear buffers on resize to make sure that the black stripes left/right
+            # are black and not polluted from previous frames. Make sure this is applied on
+            # the whole window and not within glViewport!
+            gl_utils.glClear(gl_utils.GL_COLOR_BUFFER_BIT)
+            gl_utils.glClearColor(0, 0, 0, 1)
```

Comment:
This comment is copy-pasted from PIMonitor and does not make sense here since we don't keep the viewport in a fixed aspect ratio (and thus don't have black stripes). I'm not even sure the clearing is required here? Is this just a copy-paste or is this required?

Reply:
Clearing the buffer is probably necessary as `consume_events_and_render_buffer()` is called afterward. If we do not clear the buffer we will probably see content being drawn on top of each other.

Reply:
Shouldn't this snippet then be in all UIs? I see the glClear only in world.py currently.

Reply:
Also we don't have `consume_events_and_render_buffer()` in service_ui, but in all other launchables. Maybe we should make this consistent?

Reply:
Without this snippet, you get some overlap glitching while resizing, but this disappears once you stop dragging the borders around. This also what happens currently in eye, service, and player anyways, as the clear is missing there.

With it, the overlap glitches are just black.

Reply:
@papr should we get rid of this or add it to all launchables?

Reply:
@pfaion Add it to all launchables